hayeon/programmers-1/숫자-문자열과-영단어.js

```diff
@@ -0,0 +1,37 @@
+function solution(s) {
+  const strings = [
+    "zero",
+    "one",
+    "two",
+    "three",
+    "four",
+    "five",
+    "six",
+    "seven",
+    "eight",
+    "nine",
+  ];
+
+  let result = s;
+
+  for (let i = 0; i < strings.length; i++) {
+    let changeNumber = result.split(strings[i]);
+
+    result = changeNumber.join(i);
+  }
+
+  return parseInt(result);
+}
```

Comment:
```suggestion
function solution(s) {
  const strings = [
    "zero",
    "one",
    "two",
    "three",
    "four",
    "five",
    "six",
    "seven",
    "eight",
    "nine",
  ];

  let result = s;

  for (let i = 0; i < strings.length; i++) {
    const changeNumber = result.split(strings[i]);

    result = changeNumber.join(i);
  }

  return parseInt(result);
}
```
- 풀이잘봤습니다~ split과 join을 이용한 풀이 잘봤습니다. 🙇‍♂️
- changeNumber는 const로 선언해주는게 맞을 것 같습니다 :) 
- 해당 풀이는 아래와 같이 split과 join의 특징을 잘 활용한 것 같네요 !!
```js
1two34five.split("two"); // ["1", "34five"]
["1", "34five"].join("2"); // 1234five
```
- 아래 풀이는 정규표현식으로 접근한 풀이입니다
```js
function solution(s) {
  const numberMappingObj = {
    zero: 0,
    one: 1,
    two: 2,
    three: 3,
    four: 4,
    five: 5,
    six: 6,
    seven: 7,
    eight: 8,
    nine: 9,
  };
  const regex = /(zero|one|two|three|four|five|six|seven|eight|nine)/g;

  return +s.replace(regex, (el) => numberMappingObj[el]);
}
```

Reply:
넵넵 원리는 이해했습니당!! 친절하게 설명해주셔서 감사합니다ㅎㅎ  정규표현식으로 풀이한 방법도 참고해서 한번더 풀어보겠습니닷!!😆